proposals/0026-hlsl-long-vector-type.md

```diff
@@ -54,29 +55,52 @@ Declarations of long vectors require the use of the template declaration.
 Unlike vector sizes between 1 and 4, no shorthand declarations that concatenate
  the element type and number of elements (e.g. float2, double4) are allowed for long vectors.
 
+#### Allowed Usage
+
 The new vectors will be supported in all shader stages including Node shaders. There are no control flow or wave
 uniformity requirements, but implementations may specify best practices in certain uses for optimal performance.
 
+Long vectors can be:
+
+* Elements of arrays, structs, StructuredBuffers, and ByteAddressBuffers.
+* Parameters and return types of non-entry functions.
+* Stored in groupshared memory.
+* Static global varaibles.
+
 Long vectors are not permitted in:
 
 * Resource types other than ByteAddressBuffer or StructuredBuffer.
-* Any element of the shader's signature including entry function parameters and return types.
+* Any part of the shader's signature including entry function parameters and return types.
 * Cbuffers or tbuffers.
+* A mesh/amplification `Payload` entry parameter structure.
+* A ray tracing `Parameter`, `Attributes`, or `Payload` parameter structure.
+* A work graph record.
 
 #### Constructing vectors
 
 HLSL vectors can be constructed through initializer lists, constructor syntax initialization, or by assignment.
+Vectors can be initialized and assigned from various casting operations including scalars and arrays.
+Long vectors will maintain equivalent casting abilities.
 
 Examples:
 
-``` hlsl
-vector<uint, 5> vecA = {1, 2, 3, 4, 5};
-vector<uint, 6> vecB = vector<uint, 6>(6, 7, 8, 9, 0, 0);
+```hlsl
+vector<uint, 5> InitList = {1, 2, 3, 4, 5};
+vector<uint, 6> Construct = vector<uint, 6>(6, 7, 8, 9, 0, 0);
 uint4 initval = {0, 0, 0, 0};
-vector<uint, 8> vecC = {uint2(coord.xy), vecB};
-vector<uint, 6> vecD = vecB;
+vector<uint, 8> VecVec = {uint2(coord.xy), vecB};
+vector<uint, 6> Assigned = vecB;
+float arr[5];
+vector<float, 5> CastArr = (vector<float, 5>)arr;
+vector<float, 6> ArrScal = {arr, 7.9};
+vector<float, 10> ArrArr = {arr, arr};
+vector<float, 15> Scal = 4.2;
 ```
 
+float4 main(uint size: S) : SV_Target {
+   return (float4)arr;
+vector<uint, 8> vecC = {uint2(coord.xy), vecB};
+
```

Comment:
This is outside the ` ``` ` so doesn't format correctly.  Also looks like the `main` function is missing a close brace.  Is the `vecC` part means to be inside the function?

Reply:
Left over margin notes. I've removed them.